### DISCUSSION TITLE
Running syncthing service in user instance

### DISCUSSION THREAD

**Discussion by omnigenous** [davmac314/dinit#337] [Q&A] [answered]
I had my syncthing service in system instance but now I want to start using user instance.
Here is how working system service for syncthing looks:
```
type            = process
command         = /usr/bin/syncthing -logflags=0 -home=$HOME/.config/syncthing
load-options    = export-passwd-vars
smooth-recovery = true
logfile         = /var/log/dinit/syncthing.log
# Don't forget to change this if you're not running syncthing as root
run-as          = USER
depends-on      = network
```

So I modified it for user instance:
```
type            = process
command         = /usr/bin/syncthing -logflags=0 -home=$HOME/.config/syncthing
load-options    = export-passwd-vars
smooth-recovery = true
log-type        = buffer
depends-on      = network
```

`dinitcheck syncthing` returns:
```
Checking service: syncthing...
Checking service: network...
Unable to load service 'network': service description not found.
Performing secondary checks...
Secondary checks complete.
One or more errors/warnings issued.
```

Couldn't find information on how to approach this, do I need to create network service? What do I populate it with.

**Comment by davmac314** [comment, marked as answer]
A user service cannot reference a system service (such as `network`). The user dinit instance is completely separate from the system dinit instance.

I'm assuming that the `network` service is a service which ensures that the network is set up.

Unless there's any particular reason to think that the network might not be properly set up by the time the user instance starts, you could just remove the dependency (`depends-on = network` line).

> do I need to create network service? What do I populate it with.

You could possibly also do this, but as to what to populate it with I'm not sure. Some command that can run as a normal user and waits until the network is set up.

**Reply by omnigenous** [reply]
I went ahead and removed `depends-on` field, will report back if something breaks. Thank you!
